web-ui/src/app/components/step/mwe-results/mwe-results.component.ts

```diff
@@ -48,10 +57,30 @@ export class MweResultsComponent extends ResultsComponent {
     @Output()
     public changeRetrieveContext = new EventEmitter<boolean>();
 
+    public excludeQuery: { [key: number]: boolean } = {};
+
+    private behaviour$: BehaviorSubject<SearchBehaviour> = new BehaviorSubject({
+        supersetXpath: null,
+        expandIndex: true,
+    });
+
     toggleContext() {
         this.changeRetrieveContext.emit(!this.retrieveContext);
     }
 
+    toggleExclude(query: MweQuery, event: Event) {
+        this.excludeQuery[query.rank] = (event.target as HTMLInputElement).checked;
```

Comment:
I would expect

```typescript
this.excludeQuery[query.rank] = !this.excludeQuery[query.rank];
```

to work. This makes it less dependent on the specific realization as an HTML element currently.